src/main/java/bookstore/service/BookServiceImpl.java

```diff
@@ -10,14 +14,29 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepositoryImpl bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+
+        Book newBook = bookMapper.toModel(createBookRequestDto);
+        bookRepository.save(newBook);
+        return bookMapper.toDto(newBook);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findBookById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Entity does`t exist with id: " + id));
```

Comment:
Let's make message more specific, something like:
```suggestion
                        "Failed to find book by id: " + id));
```